### PR TITLE
Feature/create entities

### PR DESCRIPTION
**Pull request by Pavliuk13**
entities and migrations created
fake data added

### CODE REVIEW

**Comment by tatianahutii** [comment]
Could you please fix all Git hub warnings?

File: backend/EasyMeets.Core/EasyMeets.Core.DAL/Entities/AdvancedSlotSettings.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
+using EasyMeets.Core.DAL.Entities.Enums;
+
+namespace EasyMeets.Core.DAL.Entities;
+
+public class AdvancedSlotSettings : Entity<int>
```
Comment (tatianahutii) [inline backend/EasyMeets.Core/EasyMeets.Core.DAL/Entities/AdvancedSlotSettings.cs]: It would be better to do Id as long (this is for all Ids)

File: backend/EasyMeets.Core/EasyMeets.Core.DAL/Entities/AdvancedSlotSettings.cs [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+{
+    public int AvailabilitySlotId { get; set; }
+    public ActivityType ActivityType { get; set; }
+    public int Days { get; set; }
+    public DateTime StartDate { get; set; }
```
Comment (tatianahutii) [inline backend/EasyMeets.Core/EasyMeets.Core.DAL/Entities/AdvancedSlotSettings.cs]: It would be better to use DateTimeOffSet instead of DateTime (this is for all dates)

File: backend/EasyMeets.Core/EasyMeets.Core.DAL/Entities/Calendar.cs [hunk 3]
```diff
@@ -0,0 +1,12 @@
+namespace EasyMeets.Core.DAL.Entities;
+
+public class Calendar : AuditEntity<int>
+{
+    public int? AddEventsFromTeamId { get; set; }
+    public bool CheckForConflicts { get; set; }
+    
+    public User User { get; set; }
+    public Team? ImportEventsFromTeam { get; set; }
```
Comment (tatianahutii) [inline backend/EasyMeets.Core/EasyMeets.Core.DAL/Entities/Calendar.cs:16]: Don't we need Ids here?

File: backend/EasyMeets.Core/EasyMeets.Core.DAL/Entities/Common/AuditEntity/AuditEntity.cs [hunk 4]
```diff
@@ -3,6 +3,5 @@
     public abstract class AuditEntity<T> : Entity<T> where T : struct
     {
         public T? CreatedBy { get; set; }
-        public DateTime CreatedAt { get; set; }
```
Comment (tatianahutii) [inline backend/EasyMeets.Core/EasyMeets.Core.DAL/Entities/Common/AuditEntity/AuditEntity.cs:6]: Why did you delete this?

File: backend/EasyMeets.Core/EasyMeets.Core.DAL/Entities/Common/Entity.cs [hunk 5]
```diff
@@ -3,5 +3,9 @@
     public abstract class Entity<T> where T : struct
     {
         public T Id { get; set; }
+        
+        public DateTime CreatedAt { get; set; }
+        public DateTime UpdatedAt { get; set; }
```
Comment (tatianahutii) [inline backend/EasyMeets.Core/EasyMeets.Core.DAL/Entities/Common/Entity.cs]: This should be a part of the AuditEntity

File: backend/EasyMeets.Core/EasyMeets.Core.DAL/Entities/TeamMember.cs [hunk 6]
```diff
@@ -0,0 +1,15 @@
+using EasyMeets.Core.DAL.Entities.Enums;
+
+namespace EasyMeets.Core.DAL.Entities;
+
+public class TeamMember
```
Comment (tatianahutii) [inline backend/EasyMeets.Core/EasyMeets.Core.DAL/Entities/TeamMember.cs]: Why you didn't use Entity here?

File: backend/EasyMeets.Core/EasyMeets.Core.DAL/Entities/TeamMemberMeeting.cs [hunk 7]
```diff
@@ -0,0 +1,11 @@
+namespace EasyMeets.Core.DAL.Entities;
+
+public class TeamMemberMeeting
```
Comment (tatianahutii) [inline backend/EasyMeets.Core/EasyMeets.Core.DAL/Entities/TeamMemberMeeting.cs]: See previous comment

**Review by tatianahutii — CHANGES_REQUESTED** [review]

**Comment by tatianahutii** [comment]
If you change something, please update the DB schema in our README file
We need to have up to date schema there

File: backend/EasyMeets.Core/EasyMeets.Core.DAL/Entities/TeamMember.cs (hunk 6 above)
Comment (Pavliuk13) [reply backend/EasyMeets.Core/EasyMeets.Core.DAL/Entities/TeamMember.cs]: we have composite primary key for this table and we don't need Entity here. Or it would be better to use Entity and make Id-field primary key ?

Comment (tatianahutii) [reply backend/EasyMeets.Core/EasyMeets.Core.DAL/Entities/TeamMember.cs]: I believe it would be better to have Id here as a combined primary key can cause some issues in the future
As specially when you have IsDeleted flag

File: backend/EasyMeets.Core/EasyMeets.Core.DAL/Entities/User.cs [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+    public string? TimeZone { get; set; }
+    public bool IsBanned { get; set; }
+
+    public ICollection<TeamMember> TeamMembers { get; set; }
+    public ICollection<Calendar> Calendars { get; set; }
+    public ICollection<Meeting> CreatedMeetings { get; set; }
+    public ICollection<TeamMemberMeeting> TeamMeetings { get; set; }
+    public ICollection<UserSlot> Slots { get; set; }
+    public ICollection<AvailabilitySlot> CreatedSlots { get; set; }
+}
```
Comment (tatianahutii) [inline backend/EasyMeets.Core/EasyMeets.Core.DAL/Entities/User.cs:29]: I still see a lot of warnings
Could you please fix them?

**Review by tatianahutii — CHANGES_REQUESTED** [review]

File: backend/EasyMeets.Core/EasyMeets.Core.DAL/Entities/Calendar.cs (hunk 3 above)
Comment (Pavliuk13) [reply backend/EasyMeets.Core/EasyMeets.Core.DAL/Entities/Calendar.cs:16]: UserId added
AddEventsFromTeamId was already exist

**Review by tatianahutii — APPROVED** [review]
LGTM
